apps/kyberswap-interface/src/pages/Earns/types.ts

```diff
@@ -77,54 +79,143 @@ export interface ParsedEarnPool extends EarnPool {
+export interface UserPositionsApiResponse {
+  code: number
+  message: string
+  data: {
+    positions: UserPosition[]
+    stats: UserPositionsStats
+  }
+  requestId: string
+}
+
+export interface UserPositionsStats {
+  totalItems: number
+  totalValueUsd: number
+  totalClaimedFeeUsd: number
+  totalUnclaimedFeeUsd: number
+  totalClaimedRewardUsd: number
+  totalUnclaimedRewardUsd: number
+  totalPendingRewardUsd: number
+}
+
+export interface UserPosition {
+  chain: {
+    name: string
+    logo: string
+    id: number
+  }
+  tokenId: number
   tokenAddress: string
-  tokenId: string
-  minPrice: number
-  maxPrice: number
-  currentAmounts: Array<PositionAmount>
-  feePending: Array<PositionAmount>
-  feesClaimed: Array<PositionAmount>
-  createdTime: number
-  stats: {
-    apr: PoolAprInterval
-    kemLMApr: PoolAprInterval
-    kemEGApr: PoolAprInterval
-    earning: PoolAprInterval
+  positionId: string
+  wallet: string
+  liquidity: string
+  status: string
+  stats: PositionStats
+  currentAmounts: TokenAmount[]
+  providedAmounts: TokenAmount[]
+  pool: PositionPool
+  suggestionPool: SuggestedPool | null
+  valueInUSD: number
+  positionCreatedTimestamp: number
+  positionCreatedBlock: number
+  lastUpdatedAt: number
+  extra: {
+    priceRange: {
+      min: number
+      maxPrice: number
+    }
   }
-  /** @deprecated */
-  apr: number
-  /** @deprecated */
-  kemEGApr: number
-  /** @deprecated */
-  kemLMApr: number
-  /** @deprecated */
-  earning24h: number
-  /** @deprecated */
-  earning7d: number
-  currentPositionValue: number
-  status: PositionStatus
-  pool: {
-    id: string
-    poolAddress: string
-    price: number
-    tokenAmounts: Array<PositionAmount>
-    fees: Array<number>
-    tickSpacing: number
-    exchange: Exchange
-    projectLogo: string
-    category: PAIR_CATEGORY
-    programs?: Array<ProgramType>
-    merklOpportunity?: MerklOpportunity
+  id: number
+}
+
+export interface PositionStats {
+  apr: {
+    all: TimeIntervalValues
+    reward: {
+      lm: TimeIntervalValues
+      eg: TimeIntervalValues
+    }
+    lp: TimeIntervalValues
+  }
+  earning: {
+    totalUsd: TimeIntervalValues
+    fee: {
+      unclaimed: TokenAmount[]
+      claimed: TokenAmount[]
+    }
+    reward: any | null
+  }
+}
+
+export interface TimeIntervalValues {
+  '24h': number
+  '7d': number
+  '30d': number
+}
+
+export interface TokenAmount {
+  amount: {
+    usdValue: number
+    priceUsd: number
+    amount: string
+  }
+  token: {
+    logo: string
+    symbol: string
+    name: string
+    decimals: number
+    address: string
+  }
+}
+
+export interface PositionPool {
+  id: string
+  address: string
+  price: number
+  tokenAmounts: TokenAmount[]
+  fees: number[]
+  programs: string[]
+  tickSpacing: number
+  protocol: {
+    type: string
+    logo: string
+    name: string
+  }
+  category: string
+  hooks: string
+  merklOpportunity?: MerklOpportunity
+}
+
+export interface EarnPosition {
```

Comment:
@linhbkhn95 this is same as UserPosition above, right?

Reply:
Yeap. l updated